src/common/image.c

```diff
@@ -1680,8 +1680,8 @@ int32_t dt_image_rename(const int32_t imgid, const int32_t filmid, const gchar *
         (dt_database_get(darktable.db),
          "SELECT id"
          " FROM main.images"
-         " WHERE filename IN (SELECT filename FROM main.images WHERE id = ?1)"
-         "   AND film_id IN (SELECT film_id FROM main.images WHERE id = ?1)",
+         " WHERE filename = (SELECT filename FROM main.images WHERE id = ?1)"
```

Comment:
isn't this just equivalent to:

`WHERE filename = ?1 AND film_id = ?2`

Or I'm confused this morning ?

Reply:
This is about duplicates. Searching for all images with the same name in the same folder.

The original query seems ok for  me.